### ISSUE TITLE
Integration tests should be structured better to support accounts that have SnapLinks disabled

### ISSUE THREAD

**Issue by dekobon** [joyent/java-manta#480]
When running just MantaClientIT, the following tests fail due to snaplinks being disabled:
```
MantaClientIT.canMoveDirectoryWithContents
MantaClientIT.canMoveFileToDifferentPrecreatedDirectory
MantaClientIT.canMoveFileToDifferentUncreatedDirectoryCreationEnabled
MantaClientIT.testHead
MantaClientIT.testPutJsonLink
MantaClientIT.testPutLink
```

We should probably have a separate integration test class that does only operations related to snaplinks and remove all tests that use snaplinks from MantaClientIT.

**Comment by nairashwin952013** [comment]
We have to hash this out probably in a meeting so that the fix for #483 and this issue can be done with the same working branch.

**Comment by nairashwin952013** [comment]
This issue has been fixed by #496  which will be merged into subsequent release version after ```3.4.0```.
